src/library/scala/collection/package.scala

```diff
@@ -77,4 +77,8 @@ package object collection {
       if(t.isEmpty) None
       else Some(t.init -> t.last)
   }
+
+  /** Common message for IndexOutOfBounds exceptions. */
+  private[collection] def indexOutOfBoundsMessage(index: Int, max: Option[Int]): String = 
```

Comment:
If the point is to reduce code size at call site, using an `Option` argument, and using a default value, are counterproductive techniques. Use overloads instead, with a bare `Int` second argument.

Reply:
Thanks, that's much better.

For reference, I also tried a sentinel value (max < 0 would be unknown, like `knownSize`), but that is not trivial without changing the current behavior:
- Some operation can actually return `max -1` (e.g. `(5 to 2)(0)`)
- Some operations error with `max ${knownSize}` while others print `max ${knownSize-1}`, so I can't just change the parameter to `maxSize`

Reply:
I see what you mean now: inclusive range is assuming closed interval. But that's just semantics. In this context, `max` is the limit of an open interval (exclusive range).